dataplane/forwarding/fwdbuilder/action.go

```diff
@@ -0,0 +1,201 @@
+		ActionType: ab.adb.atype(),
+	}
+	ab.adb.set(a)
+	return a
+}
+
+// UpdateActionBuilder is a builder for an update action.
+type UpdateActionBuilder struct {
+	fieldIDNum fwdpb.PacketFieldNum
+	updateType fwdpb.UpdateType
+	fieldSrc   fwdpb.PacketFieldNum
+	value      []byte
+}
+
+// UpdateAction returns a new update action builder.
+func UpdateAction(t fwdpb.UpdateType, num fwdpb.PacketFieldNum) *UpdateActionBuilder {
+	return &UpdateActionBuilder{
+		fieldIDNum: num,
+		updateType: t,
+	}
+}
+
+// WithFieldIDNum sets the packet field id enum.
+func (u *UpdateActionBuilder) WithFieldIDNum(num fwdpb.PacketFieldNum) *UpdateActionBuilder {
+	u.fieldIDNum = num
+	return u
+}
+
+// WithUpdateType sets the update type.
+func (u *UpdateActionBuilder) WithUpdateType(t fwdpb.UpdateType) *UpdateActionBuilder {
+	u.updateType = t
+	return u
+}
+
+// WithUpdateType sets the update source field id.
+func (u *UpdateActionBuilder) WithFieldSrc(num fwdpb.PacketFieldNum) *UpdateActionBuilder {
+	u.fieldSrc = num
+	return u
+}
+
+// WithUint64Value sets the value to a uint64 value.
+func (u *UpdateActionBuilder) WithUint64Value(v uint64) *UpdateActionBuilder {
+	u.value = binary.BigEndian.AppendUint64(nil, v)
+	return u
+}
+
+// WithUint64Value sets the value to a byte slice.
+func (u *UpdateActionBuilder) WithValue(v []byte) *UpdateActionBuilder {
+	u.value = v
+	return u
+}
+
+func (u *UpdateActionBuilder) set(ad *fwdpb.ActionDesc) {
+	upd := &fwdpb.ActionDesc_Update{
+		Update: &fwdpb.UpdateActionDesc{
+			FieldId: &fwdpb.PacketFieldId{
+				Field: &fwdpb.PacketField{
+					FieldNum: u.fieldIDNum,
+				},
+			},
+			Type:  u.updateType,
+			Value: u.value,
+			Field: &fwdpb.PacketFieldId{
+				Field: &fwdpb.PacketField{
+					FieldNum: u.fieldSrc,
+				},
+			},
+		},
+	}
+	ad.Action = upd
+}
+
+func (u *UpdateActionBuilder) atype() fwdpb.ActionType {
+	return fwdpb.ActionType_ACTION_TYPE_UPDATE
+}
+
+// TransmitActionBuilder is a builder for a transmit action.
+type TransmitActionBuilder struct {
+	portID string
+}
+
+// TransmitAction returns a new update action builder.
+func TransmitAction(portID string) *TransmitActionBuilder {
+	return &TransmitActionBuilder{
+		portID: portID,
+	}
+}
+
+// WithFieldIDNum sets thje port id value.
+func (u *TransmitActionBuilder) WithPortID(id string) *TransmitActionBuilder {
+	u.portID = id
+	return u
+}
+
+func (u *TransmitActionBuilder) set(ad *fwdpb.ActionDesc) {
+	upd := &fwdpb.ActionDesc_Transmit{
+		Transmit: &fwdpb.TransmitActionDesc{
+			PortId: &fwdpb.PortId{
+				ObjectId: &fwdpb.ObjectId{
+					Id: u.portID,
+				},
+			},
+		},
+	}
+	ad.Action = upd
+}
+
+func (u *TransmitActionBuilder) atype() fwdpb.ActionType {
+	return fwdpb.ActionType_ACTION_TYPE_TRANSMIT
+}
+
+// TransmitActionBuilder is a builder for a transmit action.
+type LookupActionBuilder struct {
+	tableID string
+}
+
+// TransmitAction returns a new update action builder.
+func LookupAction(tableID string) *LookupActionBuilder {
```

Comment:
NewLookupAction?

Reply:
following style of link in earlier, i think it's clear without the New prefix

Reply:
I think if the package name were `fwdconfig.LookupAction` then it is more clear. I think it's just the "builder" part that's confusing me since it's not the conventional builder type I was expecting.

Reply:
renamed